packages/@apphosting/adapter-nextjs/src/overrides.ts

```diff
@@ -0,0 +1,55 @@
+import { AdapterMetadata, MiddlewareManifest } from "./interfaces.js";
+import { loadRouteManifest, writeRouteManifest, loadMiddlewareManifest } from "./utils.js";
+
+/**
+ * Modifies the app's route manifest (routes-manifest.json) to add Firebase App Hosting
+ * specific overrides (i.e headers).
+ *
+ * This function adds the following headers to all routes:
+ * - x-fah-adapter: The Firebase App Hosting adapter version used to build the app.
+ * - x-fah-middleware: When middleware is enabled.
+ *
+ * @param appPath The path to the app directory.
+ * @param distDir The path to the dist directory.
+ * @param adapterMetadata The adapter metadata.
+ */
+export async function addRouteOverrides(
+  appPath: string,
+  distDir: string,
+  adapterMetadata: AdapterMetadata,
+) {
+  const middlewareManifest = loadMiddlewareManifest(appPath, distDir);
+  const routeManifest = loadRouteManifest(appPath, distDir);
+  routeManifest.headers.push({
+    source: "/:path*",
+    headers: [
+      {
+        key: "x-fah-adapter",
+        value: `nextjs-${adapterMetadata.adapterVersion}`,
+      },
+      ...(middlewareExists(middlewareManifest)
+        ? [
+            {
+              key: "x-fah-middleware",
+              value: "true",
+            },
+          ]
+        : []),
+    ],
+    /* 
+      NextJs converts the source string to a regex using path-to-regexp (https://github.com/pillarjs/path-to-regexp) at 
+      build time: https://github.com/vercel/next.js/blob/canary/packages/next/src/build/index.ts#L1273.
+      This regex is then used to match the route against the request path.
+
+      This regex was generated by building a sample NextJs app with the source string `/:path*` and then inspecting the
+      routes-manifest.json file.
+    */
+    regex: "^(?:/((?:[^/]+?)(?:/(?:[^/]+?))*))?(?:/)?$",
```

Comment:
btw you can use `path-to-regex` to generate this rather than hardcode

`> pathToRegexp("/:path*") => /^(?:\/((?:[^\/]+?)(?:\/(?:[^\/]+?))*))?(?:\/)?$/i`

Reply:
oh see you have a comment explaining that now, heh